packages/app/src/cli/models/extensions/specifications/flow_action.ts

```diff
@@ -0,0 +1,50 @@
+import {BaseSchema} from '../schemas.js'
+import {createExtensionSpecification} from '../specification.js'
+
+import {zod} from '@shopify/cli-kit/node/schema'
+
+const FlowActionExtensionSchema = BaseSchema.extend({
+  name: zod.string(),
+  type: zod.literal('flow_action'),
+  task: zod.object({
+    title: zod.string(),
+    description: zod.string(),
+    url: zod.string(),
+    validationUrl: zod.string().optional(),
+    customConfigurationPageUrl: zod.string().optional(),
+    customConfigurationPagePreviewUrl: zod.string().optional(),
+    fields: zod.array(
```

Comment:
Currently we can create an action with no fields - should this array be optional?

Reply:
Ah yes good call.